include/wlr/types/wlr_keyboard_shortcuts_inhibit_v1.h

```diff
@@ -0,0 +1,103 @@
+/*
+ * This an unstable interface of wlroots. No guarantees are made regarding the
+ * future consistency of this API.
+ */
+#ifndef WLR_USE_UNSTABLE
+#error "Add -DWLR_USE_UNSTABLE to enable unstable wlroots features"
+#endif
+
+#ifndef WLR_TYPES_WLR_KEYBOARD_SHORTCUTS_INHIBIT_V1_H
+#define WLR_TYPES_WLR_KEYBOARD_SHORTCUTS_INHIBIT_V1_H
+
+#include <wayland-server-core.h>
+#include <wlr/types/wlr_seat.h>
+
+/* This interface permits clients to inhibit keyboard shortcut processing by
+ * the compositor.
+ *
+ * This allows clients to pass them on to e.g. remote desktops or virtual
+ * machine guests.
+ *
+ * Inhibitors are created for surfaces and seats. They should only be in effect
+ * while this surface has focus.
+ */
+
+struct wlr_keyboard_shortcuts_inhibit_manager_v1 {
+	// wlr_keyboard_shortcuts_inhibitor_v1::link
+	struct wl_list inhibitors;
+	struct wl_global *global;
+
+	struct wl_listener display_destroy;
+
+	struct {
+		struct wl_signal new_inhibitor;	// wlr_keyboard_shortcuts_inhibitor_v1
+		struct wl_signal destroy;
+	} events;
+
+	void *data;
+};
+
+struct wlr_keyboard_shortcuts_inhibitor_v1 {
+	struct wlr_surface *surface;
+	struct wlr_seat_client *seat_client;
+	bool active;
+	struct wl_resource *resource;
+	struct wl_listener surface_destroy;
```

Comment:
We probably also need a listener of the seat client destruction?

Reply:
Right and done.

Aside: Is seat client a good choice here? It was the one seat type object I could find a function for to turn up a reference on from a wayland seat resource. The inhibitor could just as well just remember the seat which would be closer to the protocol spec and likely still provide the compositor with all it needs to know. I've got no idea of the repercussions either way.

Reply:
Yeah, I think using the seat would be better, since it'll still work if the client destroys all of its `wl_seat` resources.

Reply:
Done.